client/e2e/support/add-user.po.ts

```diff
@@ -0,0 +1,47 @@
+export interface TestUser {
+  name: string;
+  age: string;
+  company?: string;
+  email?: string;
+  role: 'admin' | 'editor' | 'viewer';
+}
+
+export class AddUserPage {
+  navigateTo() {
+    return cy.visit('/users/new');
+  }
+
+  getTitle() {
+    return cy.get('.add-user-title');
+  }
+
+  addUserButton() {
+    return cy.contains('button', 'ADD USER');
+  }
+
+  typeInput(inputId: string, text: string) {
+    const input = cy.get(`#${inputId}`).click();
+    input.type(text);
+  }
+
+  selectMatSelectValue(selectID: string, value: string) {
+    const sel = cy.get(`#${selectID}`).click().get(`mat-option[value="${value}"]`).click();
```

Comment:
This does a _lot_ of stuff and is not at all obvious to read (at least not to me). We almost certainly want a comment or two, and we might want to break it up into multiple (lexical) lines (if not semantic lines).

Reply:
To add to that, why are we setting it as a `const` value and not returning the result?